src/sentry/web/frontend/base.py

```diff
@@ -294,78 +326,84 @@ def dispatch(self, request, *args, **kwargs):
         return (args, kwargs)
 
-    def handle(self, request: Request, *args, **kwargs) -> Response:
+    def handle(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         return super().dispatch(request, *args, **kwargs)
 
-    def is_auth_required(self, request: Request, *args, **kwargs):
+    def is_auth_required(self, request: Request, *args: Any, **kwargs: Any) -> bool:
         return self.auth_required and not (request.user.is_authenticated and request.user.is_active)
 
-    def handle_auth_required(self, request: Request, *args, **kwargs):
+    def handle_auth_required(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
         auth.initiate_login(request, next_url=request.get_full_path())
         if "organization_slug" in kwargs:
             redirect_to = reverse("sentry-auth-organization", args=[kwargs["organization_slug"]])
         else:
             redirect_to = auth.get_login_url()
         return self.redirect(redirect_to, headers={"X-Robots-Tag": "noindex, nofollow"})
 
-    def is_sudo_required(self, request: Request, *args, **kwargs):
+    def is_sudo_required(self, request: Request, *args: Any, **kwargs: Any) -> bool:
         return self.sudo_required and not request.is_sudo()
 
-    def handle_sudo_required(self, request: Request, *args, **kwargs):
-        return redirect_to_sudo(request.get_full_path())
+    def handle_sudo_required(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
+        return redirect_to_sudo(request.get_full_path())  # type: ignore
 
-    def has_permission(self, request: Request, *args, **kwargs):
+    def has_permission(self, request: Request, *args: Any, **kwargs: Any) -> bool:
         return True
 
-    def handle_permission_required(self, request: Request, *args, **kwargs):
+    def handle_permission_required(
+        self, request: Request, *args: Any, **kwargs: Any
+    ) -> HttpResponse:
         redirect_uri = self.get_no_permission_url(request, *args, **kwargs)
         return self.redirect(redirect_uri)
 
-    def handle_not_2fa_compliant(self, request: Request, *args, **kwargs):
+    def handle_not_2fa_compliant(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
         redirect_uri = self.get_not_2fa_compliant_url(request, *args, **kwargs)
         return self.redirect(redirect_uri)
 
-    def get_no_permission_url(self, request: Request, *args, **kwargs):
-        return reverse("sentry-login")
+    def get_no_permission_url(self, request: Request, *args: Any, **kwargs: Any) -> str:
+        return cast(str, reverse("sentry-login"))
 
-    def get_not_2fa_compliant_url(self, request: Request, *args, **kwargs):
-        return reverse("sentry-account-settings-security")
+    def get_not_2fa_compliant_url(self, request: Request, *args: Any, **kwargs: Any) -> str:
+        return cast(str, reverse("sentry-account-settings-security"))
```

Comment:
Alternatively, you might consider adding `# type: ignore` instead of casting when calling untyped modules ala: https://github.com/getsentry/sentry/pull/40795#pullrequestreview-1163592952